agenta-backend/agenta_backend/services/app_manager.py

```diff
@@ -197,8 +197,13 @@ async def remove_app_variant(
                 await db_manager.remove_image(image, **kwargs)
 
                 # Only delete the docker image for users that are running the oss version
-                if os.environ["FEATURE_FLAG"] not in ["cloud", "ee", "demo"]:
-                    docker_utils.delete_image(image.docker_id)
+                try:
+                    if os.environ["FEATURE_FLAG"] not in ["cloud", "ee", "demo"]:
+                        docker_utils.delete_image(image.docker_id)
+                except Exception as e:
+                    logger.error(
+                        f"Ignoring error while deleting Docker image: {str(e)}"
```

Comment:
Why do we have an ignore in this case?

Reply:
because an image can have running containers other than the current variant. In this case, an exception is thrown, preventing the app from deletion.